src/core/Entry.cpp

```diff
@@ -778,17 +778,30 @@ QString Entry::resolvePlaceholderRecursive(const QString& placeholder, int maxDe
     case PlaceholderType::Unknown:
         return placeholder;
     case PlaceholderType::Title:
-        return title();
+        if (placeholderType(title()) == typeOfPlaceholder) {
```

Comment:
Took me a while to understand what you are doing here. You seem to be short cutting infinitely recursing self-references. I would prefer you use an explicit value on the right-hand side instead of `typeOfPlaceholder`. Makes the whole thing a lot easier to read.

Also: `maxDepth` isn't used at all. You need to check that to cancel the recursion and return the current value as-is when the counter hits 0 (then you also don't necessarily need the shortcut above, but you may keep it anyway, just make it easier to understand).

Reply:
I know about `maxDepth`, it was a 5 minute fix to avoid delaying the beta 2.3.0 release.

Like I said 
> The code for reference resolving is very repetitive, a refactor need to take place in next version.

Reply:
Just check it maxDepth is 0 and then return the value unchanged. Much easier than everything else in this PR.